packages/react-link/src/components/Link/useLinkState.ts

```diff
@@ -25,38 +27,39 @@ export const useLinkState = (state: LinkState): LinkState => {
   // Adjust props depending on the root type.
   if (typeof as === 'string') {
     if (as === 'a') {
-      state.href = disabled ? undefined : href;
-      state.tabIndex = disabled && !disabledFocusable ? undefined : 0;
+      state.root;
+      state.root.href = disabled ? undefined : href;
+      state.root.tabIndex = disabled && !disabledFocusable ? undefined : 0;
     }
     // Remove the href, rel and target props for all non-anchor elements.
     else {
-      state.href = undefined;
-      state.rel = undefined;
-      state.target = undefined;
+      state.root.href = undefined;
```

Comment:
it seems that this logic could be moved to something like `useAriaLink`  hook similar to the `useAriaButton`  hook that @bsunderhus  implemented

Reply:
It makes sense, I'll make the change in a separate PR.

Reply:
Added issue here #19816